tchannel/tornado/peer.py

```diff
@@ -618,7 +617,8 @@ def send_with_retry(self, request, peer, retry_limit, connection):
             try:
                 response = yield self._send(connection, request)
                 raise gen.Return(response)
-            except (ProtocolError, TimeoutError) as error:
+            # Why are we retying on all errors????
+            except TChannelError as error:
```

Comment:
I see what you're saying, yeah we don't need to do this `should_retry_on_error` stuff anymore.